### PR TITLE
Added zod validation for adding article series

### PR DESCRIPTION
**Pull request by mixamum**

### CODE REVIEW

File: packages/server/src/api/index.ts [hunk 1]
```diff
@@ -488,13 +488,11 @@ router.delete('/user/article_series/:id/articles', async (req, res) => {
...
 router.post('/user/article_series', async (req, res) => {
     // title is required
-    if (!('title' in req.body)) {
-        res.status(400).send({ error: 'Title is required' });
-        return;
-    }
 
-    if (!req.body.title) {
-        res.status(400).send({ error: 'Title is required' });
+    const response = titleValidator.safeParse({ title: req.body.title });
```
Comment (prasenjeet-symon) [inline packages/server/src/api/index.ts:492]: Simply pass the req.body . it does contain title

**Review by prasenjeet-symon — CHANGES_REQUESTED** [review]
Need a little bit of changes as commented.

File: packages/server/src/api/index.ts (hunk 1 above)
Comment (mixamum) [reply packages/server/src/api/index.ts:492]: I tried that, it always puts out "Title must be a string" even if it is passed in as a string

Comment (prasenjeet-symon) [reply packages/server/src/api/index.ts:492]: Try this one 

```js
export const titleValidator = z.object({
    title: z
        .string()
        .min(1, {
            message: "Title must be at least 1 character long",
        })
        .or(z.undefined(), {
            message: "Title is required",
        }),
});
```

Comment (mixamum) [reply packages/server/src/api/index.ts:492]: Still same thing but now "invalid input"

Comment (mixamum) [reply packages/server/src/api/index.ts]: Got it, was passing it wrong

**Review by prasenjeet-symon — APPROVED** [review]
Awesome 👍

**Comment by prasenjeet-symon** [comment]
In the commit message mention the issue number also from now on like fixed #issuenumber . Thanks

**Pull request merged**